parsers/jscript.c

```diff
@@ -745,7 +745,10 @@ static void skipArgumentList (tokenInfo *const token, boolean include_newlines,
 		while (nest_level > 0 && ! isType (token, TOKEN_EOF))
 		{
 			readTokenFull (token, FALSE, repr);
-			if (isType (token, TOKEN_OPEN_PAREN))
+			if (isType (token, TOKEN_KEYWORD) && token->keyword == KEYWORD_function &&
+			    repr == NULL)
```

Comment:
not handling in case of `repr` makes the behavior probably a little too unpredictable.  Is there a reason not to, apart missing stuff in the `repr`?  Does it lead to some problem?